admin/easypopulate_4.php

```diff
@@ -213,20 +213,20 @@
 /* @var $epdlanguage_query type array */
 //$epdlanguage_query = $db->Execute("SELECT languages_id, name FROM ".TABLE_LANGUAGES." WHERE code = '".DEFAULT_LANGUAGE."'");
 if (!defined(DEFAULT_LANGUAGE)) {
-  $epdlanguage_query = ep_4_query("SELECT languages_id, code FROM " . TABLE_LANGUAGES . " ORDER BY languages_id LIMIT 1");
-  $epdlanguage = ($ep_uses_mysqli ? mysqli_fetch_array($epdlanguage_query) : mysql_fetch_array($epdlanguage_query));
-  define('DEFAULT_LANGUAGE', $epdlanguage['code']);
+  $epdlanguage_query = "SELECT languages_id, code FROM " . TABLE_LANGUAGES . " ORDER BY languages_id LIMIT 1";
+  $epdlanguage = $db->Execute($epdlanguage_query);
+  define('DEFAULT_LANGUAGE', $epdlanguage->fields['code']);
 }
-$epdlanguage_query = ep_4_query("SELECT languages_id, name FROM " . TABLE_LANGUAGES . " WHERE code = '" . DEFAULT_LANGUAGE . "'");
-if (($ep_uses_mysqli ? mysqli_num_rows($epdlanguage_query) : mysql_num_rows($epdlanguage_query))) {
-  $epdlanguage = ($ep_uses_mysqli ? mysqli_fetch_array($epdlanguage_query) : mysql_fetch_array($epdlanguage_query));
-  $epdlanguage_id = $epdlanguage['languages_id'];
-  $epdlanguage_name = $epdlanguage['name'];
+$epdlanguage_query = "SELECT languages_id, name FROM " . TABLE_LANGUAGES . " WHERE code = '" . DEFAULT_LANGUAGE . "'";
+$epdlanguage = $db->Execute($epdlanguage_query);
+if ($epdlanguage->RecordCount() > 0) {
+  $epdlanguage_id = $epdlanguage->fields['languages_id'];
+  $epdlanguage_name = $epdlanguage->fields['name'];
 } else {
   exit("EP4 FATAL ERROR: No default language set."); // this should never happen
 }
 
-$langcode = ep_4_get_languages(); // array of currently used language codes ( 1, 2, 3, ...)
```

Comment:
This style of query offers in house error management and does not crash to resolve the issue by use of the ZC /cache or /logs directory or some other "outside" location, but instead provide a debug file that is readily available on screen for review.  This and other similar potential replacements were considered in recent upgrade of the code.  That said the below manufacturers_query was not "refactored" to use this style of query last year in light of other issues trying to be addressed at the time the most of which was to ensure compatibility with ZC 1.5.4.